### PR TITLE
add more doctests

### PR DESCRIPTION
**Pull request by KristofferC**

### CODE REVIEW

File: base/complex.jl [hunk 1]
```diff
@@ -19,8 +19,13 @@ Complex(x::Real) = Complex(x, zero(x))
     im
 
 The imaginary unit.
+
+# Example
+```jldoctest
+julia> im * im
+-1 + 0im
```
Comment (tkelman) [inline base/complex.jl]: missing closing backticks. did you run these?

File: base/iterators.jl [hunk 2]
```diff
@@ -267,6 +267,16 @@ returned iterable object. These calls are not cached and repeated calls will be
 made when reiterating.
 
 See [`Base.filter`](@ref) for an eager implementation of filtering for arrays.
+
+```
```
Comment (tkelman) [inline base/iterators.jl]: jldoctest, example heading

File: base/random.jl [hunk 3]
```diff
@@ -1389,7 +1424,7 @@ end
 Fill the array `A` with normally-distributed (mean 0, standard deviation 1) random numbers.
 Also see the [`rand`](@ref) function.
 
-# Example
+# Examples
```
Comment (tkelman) [inline base/random.jl]: not really plural

File: base/random.jl [hunk 4]
```diff
@@ -1478,7 +1513,7 @@ Generates a version 1 (time-based) universally unique identifier (UUID), as spec
 by RFC 4122. Note that the Node ID is randomly generated (does not identify the host)
 according to section 4.5 of the RFC.
 
-# Example
+# Examples
```
Comment (tkelman) [inline base/random.jl]: not plural

File: base/reflection.jl [hunk 5]
```diff
@@ -127,6 +131,7 @@ fieldname(t::Type{<:Tuple}, i::Integer) = i < 1 || i > nfields(t) ? throw(Bounds
 
 Get an array of the fields of a `DataType`.
 
+# Examples
```
Comment (tkelman) [inline base/reflection.jl]: singular

File: base/reflection.jl [hunk 6]
```diff
@@ -894,6 +948,7 @@ end
 Determine whether the given generic function has a method matching the given
 `Tuple` of argument types with the upper bound of world age given by `world`.
 
+# Examples
```
Comment (tkelman) [inline base/reflection.jl]: singular

File: base/util.jl [hunk 7]
```diff
@@ -819,13 +820,19 @@ expression. The default argument is supplied by declaring fields of the form `fi
...
 function.
 
-```julia
-@kwdef struct Foo
-    a::Cint            # implied default Cint(0)
-    b::Cint = 1        # specified default
-    z::Cstring         # implied default Cstring(C_NULL)
-    y::Bar             # implied default Bar()
-end
+```jldoctest
```
Comment (tkelman) [inline base/util.jl]: # Example

File: base/util.jl [hunk 8]
```diff
@@ -878,6 +885,22 @@ The default value for a type for use with the `@kwdef` macro. Returns:
  - null pointer for pointer types (`Ptr{T}`, `Cstring`, `Cwstring`)
  - zero for integer types
  - no-argument constructor calls (e.g. `T()`) for all other types
+
+```jldoctest
```
Comment (tkelman) [inline base/util.jl]: # Example

**Comment by KristofferC** [comment]
Can we settle on just using `#Examples` everywhere? Sometimes it is hard to say what counts as one vs two examples.

**Comment by tkelman** [comment]
as a separate pr that applied that everywhere

File: base/complex.jl (hunk 1 above)
Comment (KristofferC) [inline base/complex.jl]: Yes, a missing closing backtick has the effect that the doctest is not run at all.

**Comment by KristofferC** [comment]
Thanks, updated. I'll leave the plural headers as is.

File: base/complex.jl (hunk 1 above)
Comment (tkelman) [inline base/complex.jl]: that sounds like a bug in documenter

**Comment by tkelman** [comment]
if you do that, you should pluralize all the newly added ones too

File: base/complex.jl (hunk 1 above)
Comment (tkelman) [inline base/complex.jl]: which should be reported there

**Comment by KristofferC** [comment]
Good to go?

File: base/iterators.jl [hunk 9]
```diff
@@ -267,6 +267,17 @@ returned iterable object. These calls are not cached and repeated calls will be
 made when reiterating.
 
 See [`Base.filter`](@ref) for an eager implementation of filtering for arrays.
+
+# Examples
+```jldoctest
+julia> f = Iterators.filter(isodd, [1, 2, 3, 4, 5])
+Base.Iterators.Filter{Base.#isodd,Array{Int64,1}}(isodd, [1, 2, 3, 4, 5])
+
+julia> for v in f
```
Comment (kshyatt) [inline base/iterators.jl]: this example might be prettier as `foreach(println, v)`?

File: base/reflection.jl [hunk 10]
```diff
@@ -455,6 +508,7 @@ end
 Return a list of immediate subtypes of DataType `T`. Note that all currently loaded subtypes
 are included, including those not visible in the current module.
 
+# Example
```
Comment (kshyatt) [inline base/reflection.jl]: Shouldn't this be `Examples`?

File: base/sparse/sparsevector.jl [hunk 11]
```diff
@@ -176,6 +177,7 @@ sparsevec(I::AbstractVector, v::Number, len::Integer, combine::Function) =
 Create a sparse vector of length `m` where the nonzero indices are keys from
 the dictionary, and the nonzero values are the values from the dictionary.
 
+# Example
```
Comment (kshyatt) [inline base/sparse/sparsevector.jl]: `Examples?`?

**Review by kshyatt — APPROVED** [review]
Modulo my nits LGTM!

**Comment by KristofferC** [comment]
Addressed comments, whitespace passes.

**Review by Sacha0 — APPROVED** [review]
lgtm! :)

**Pull request merged**